code/datums/diseases/advance/symptoms/macrophage.dm

```diff
@@ -0,0 +1,88 @@
+                      <b>Resistance:</b>The higher the resistance, the more health phages will have, and the more damage they will do.<br>\
+					  <b>Transmission 10:</b>Phages can be larger, more aggressive, and able to pierce thick clothing, with some effort.<br>\
+                      <b>Transmission 12:</b>Phages will carry all diseases within the host, instead of only diseases containing their own symptom"
+
+
+
+/datum/symptom/macrophage/severityset(datum/disease/advance/A)
+	. = ..()
+	if(A.properties["transmittable"] >= 10)
+		severity += 2
+
+/datum/symptom/macrophage/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	netspeed = max(1, (A.properties["stage_rate"]))
+	if(A.properties["transmittable"] >= 10)
+		gigagerms = TRUE
+
+/datum/symptom/macrophage/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/M = A.affected_mob
+	phagecounter -= max(1, A.properties["stage_rate"])
+	switch(A.stage)
+		if(1-3)
+			to_chat(M, "<span class='notice'>Your skin crawls.</span>")
+		if(4)
+			M.visible_message("<span class='danger'>lumps form on [M]'s skin!</span>", \
+								  "<span class='userdanger'>You cringe in pain as lumps form and move around on your skin!</span>")
+		if(5)
+			if(gigagerms && phagecounter <= 0) //only ever spawn one big germ
+				Burst(A, M, TRUE)
+				phagecounter += 5 //but big germs spawn small ones more frequently
+			while(phagecounter <= 0)
+				phagecounter += 10
+				Burst(A, M)
+
+/datum/symptom/macrophage/proc/Burst(datum/disease/advance/A, var/mob/living/M, var/gigagerms = FALSE)
+	if(gigagerms)
+		var/mob/living/simple_animal/hostile/macrophage/aggro/phage = new(M.loc)
+		phage.health += A.properties["resistance"]
+		phage.maxHealth += A.properties["resistance"]
+		phage.melee_damage_upper += max(0, A.properties["resistance"])
+		phage.melee_damage_lower += max(0, A.properties["resistance"])
+		phage.infections += A
+		phage.basedisease = A
+		if(A.properties["transmittable"] >= 12)
+			for(var/datum/disease/D in M.diseases)
+				if((D.spread_flags & DISEASE_SPREAD_SPECIAL) || (D.spread_flags & DISEASE_SPREAD_NON_CONTAGIOUS) || (D.spread_flags & DISEASE_SPREAD_FALTERED))
+					continue
+				if(D == A)
+					continue
+				phage.infections += D
+		M.take_overall_damage(brute = rand(10, 20), required_status = BODYPART_ORGANIC)
+		playsound(M, 'sound/effects/splat.ogg', 50, 1)
+		M.emote("scream")
+	else
+		var/mob/living/simple_animal/hostile/macrophage/phage = new(M.loc)
+		phage.health += A.properties["resistance"]
+		phage.maxHealth += A.properties["resistance"]
+		phage.melee_damage_upper += max(0, A.properties["resistance"])
+		phage.infections += A
+		phage.basedisease = A
+		if(A.properties["transmittable"] >= 12)
+			for(var/datum/disease/D in M.diseases)
+				if((D.spread_flags & DISEASE_SPREAD_SPECIAL) || (D.spread_flags & DISEASE_SPREAD_NON_CONTAGIOUS) || (D.spread_flags & DISEASE_SPREAD_FALTERED))
+					continue
+				if(D == A)
+					continue
+				phage.infections += D
```

Comment:
Since this code is almost the same, could you not have something along the lines of
```
var/mob/living/simple_animal/hostile/macrophage/phage
if(gigagerms)
   phage = new /mob/living/simple_animal/hostile/macrophage/aggro(M.loc)
else
   phage = new(M.loc)
```
and then there would be less repeated code

Reply:
thought of it, but i aint sure if that'd work

Reply:
The only other stuff you would have to put in the if would be the scream code, since the rest is identical

Reply:
Not quite , still doable i spose